docker/service/src/main/java/uk/nhs/adaptors/scr/controllers/validation/acs/AcsRequestValidator.java

```diff
@@ -0,0 +1,99 @@
+package uk.nhs.adaptors.scr.controllers.validation.acs;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Parameters.ParametersParameterComponent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.scr.components.FhirParser;
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.exceptions.FhirValidationException;
+import uk.nhs.adaptors.scr.models.AcsPermission;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AcsRequestValidator implements ConstraintValidator<AcsRequest, String> {
+
+    private static final String SET_PERMISSION_PARAM_NAME = "setPermissions";
+    private static final String NHS_NUMBER_PART_NAME = "nhsNumber";
+    private static final String PERMISSION_CODE_PART_NAME = "permissionCode";
+    private static final String PERMISSION_CODE_SYSTEM = "https://fhir.nhs.uk/CodeSystem/SCR-ACSPermission";
+
+    private final FhirParser fhirParser;
+
+    @Override
+    public void initialize(AcsRequest contactNumber) {
+    }
+
+    @Override
+    public boolean isValid(String requestBody, ConstraintValidatorContext context) {
+        try {
+            Parameters parameters = fhirParser.parseResource(requestBody, Parameters.class);
+
+            ParametersParameterComponent parameter = getSetPermissionParameter(parameters);
+            isNhsNumber(parameter);
```

Comment:
Let's use 'is' method prefix when method returns a boolean and 'check' when it only verifies the body and throws an exception.